### DISCUSSION TITLE
How to configure pug plugin in config file ?

### DISCUSSION THREAD

**Discussion by sera2007** [brunch/brunch#1878] [Q&A]
I'm using the code

`module.exports = { ... templates: { joinTo: 'javascripts/app.js' }, },`

`plugins: { pugjs: { staticBasedir: 'app/static_pug', staticPretty: true, compileDebug: true, }, ....`

But I don't get anything as result. I was trying to change joinTo param, but it didn't work for me. Do you have any thinking?

**Comment by denysdovhan** [comment]
@sera2007 not sure what you're talking about. Try this:

```js
module.exports = {
  files: {
    templates: { joinTo: 'javascripts/app.js' },
  },
  plugins: {
    pugjs: {
      // you options here
    }
  }
}
```

**Comment by aMarCruz** [comment]
@sera2007 , for pugjs-brunch the key is `pug`.

`staticBasedir` is the "assets" directory of Brunch (any with "assets/" by default). Pug templates here will be converted to static html **files** inside the "public" directory, so don't use `staticBasedir` unless you have special needs.

For precompilation you must to use 2 options: `preCompile` and `preCompilePattern`.
Example:

```js
module.exports = {
  files: {
    templates: { joinTo: 'javascripts/app.js' },
  },
  plugins: {
    pug: {
      preCompile: true,
      preCompilePattern: /\.html\.pug$/
    }
  }
}
```

The above settings will generate functions (in javascripts/app.js) that returns plain HTML **strings** for all files ending with ".html.pug" wherever they are --inside the "app" directory, except in "assets/". Example:
```
const tmpl = require('./views/content.html.pug')
$('#container').html(tmpl)
```
